metaflow/subprocess_manager.py

```diff
@@ -0,0 +1,125 @@
+import os
```

Comment:
I am not quite sure the log management is exactly what we want. You seem to be:
- streaming logs to a file
- reading from those files to a queue
- and then streaming all the queue out at once

Granted, it only streams up to that point but it feels a little complex. A better approach may be to read the file nad remember where you are when reading so you can then read it again starting at the right location and continuing. That would enable use to have a cheaper streaming with a lot less overhead in terms of memory.